src/workflows/airqo_etl_utils/bigquery_api.py

```
@@ -766,6 +766,123 @@ def dynamic_averaging_query(
+                    query_parts.append(
+                        f"COUNT({v}) AS sample_count_{v}, "
+                        f"MAX({v}) AS maximum_{v}, "
+                        f"MIN({v}) AS minimum_{v}, "
+                        f"AVG({v}) AS average_{v}"
+                    )
+                    pollutants_list.append(v)
+
+        query = (
+            "SELECT " + ", ".join(query_parts) + f" FROM `{table}` "
+            f"WHERE timestamp BETWEEN '{start_date_time}' AND '{end_date_time}' "
+            f"AND {unique_id} = '{filter}'"
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)